File: backend/src/main/java/com/guildnet/backend/features/notification/dto/NotificationDTO.java
package com.guildnet.backend.features.notification.dto;

import com.guildnet.backend.features.communityProfile.dto.CommunityProfileDTO;
import com.guildnet.backend.features.notification.NotificationType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class NotificationDTO {
    private Long id;
    private String message;
    private NotificationType type;
    private boolean read;
    private LocalDateTime createdAt;
    private CommunityProfileDTO receiver;
    private CommunityProfileDTO sender;
}

File: frontend/src/app/models/notification/NotificationDTO.ts
import { NotificationType } from './NotificationType';
import { CommunityProfileDTO } from '../communityProfile/CommunityProfileDTO';

export interface NotificationDTO {
  id: number;
  message: string;
  type: NotificationType;
  read: boolean;
  createdAt: string;
  receiver: CommunityProfileDTO;
  sender: CommunityProfileDTO | null;
}
